pkg/apiserver/resthandler.go

```diff
@@ -787,7 +787,15 @@ func DeleteResource(r rest.GracefulDeleter, checkBody bool, scope RequestScope,
 		if admit != nil && admit.Handles(admission.Delete) {
 			userInfo, _ := api.UserFrom(ctx)
 
-			err = admit.Admit(admission.NewAttributesRecord(nil, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Delete, userInfo))
+			var currentObject runtime.Object
```

Comment:
Storage already has the ability to do precondition checking on delete, which means it is doing a get internally. If we want this info, I'd rather plumb the admission to that point (though it's a bigger change).  otherwise there's possibility for a race and the object passed to admission here might not be the object being deleted at all